src/operator/optimizer_op-inl.h

```diff
@@ -165,69 +165,6 @@ inline void SGDUpdateDnsRspImpl(const SGDParam& param,
   });
 }
 
-/*! \brief kernel for sparse sgd
- */
-template<int req>
-struct SGDRspDnsKernel {
```

Comment:
why removing all this?

Reply:
To behave consistently with what's stated on doc, only perform sparse updates for `sgd(row_sparse_weight, row_sparse_grad)` for now. Otherwise, we also need to handle sparse weights with dense grads for Adam, Ftrl, etc. The number of cases is too large

Reply:
Shouldn't sgd(dense_weight, row_sparse_grad) be supported?
It almost never make sense for weight to be row sparse